pypeit/specobjs.py

```diff
@@ -791,6 +812,15 @@ def write_to_fits(self, subheader, outfile, overwrite=True, update_det=None,
                         header[key.upper()] = line
             else:
                 header[key.upper()] = subheader[key]
+                # Also store the datetime in ISOT format
+                if key.upper() == 'MJD':
+                    if isinstance(subheader[key], (list, tuple)):
+                        mjdval = subheader[key][0]
+                    elif isinstance(subheader[key], float):
+                        mjdval = subheader[key]
+                    else:
+                        raise ValueError('Header card must be a float or a FITS header tuple')
+                    header['DATETIME'] = (Time(mjdval, format='mjd').isot, "Date and time of the observation in ISOT format")
```

Comment:
What if this is already in the header?  I.e., doesn't the header include most/all information from the raw images?  Would it make sense to instead add a new header keyword that's specific to the data reduction?  Maybe `RDXDATE` or `PYPDATE` or something similar?